### PR TITLE
e2e testing using e2e-environment

### PR DESCRIPTION
**Pull request by rrennick**
See #2497

This PR uses `@woocommerce/e2e-environment:0.1.5` to implement the following from the master thread:

- [x] Implement initial Puppeteer architecture
- [x] Write a simple test to test Puppeteer config (Uses core test suite.)
- [x] Integrate [Gutenberg’s WP util functions package](https://github.com/WordPress/gutenberg/tree/master/packages/e2e-test-utils) (In the core component.)
- [x] Write [documentation on e2e test suite config](https://github.com/woocommerce/woocommerce/blob/master/tests/e2e/env/README.md)
- [x] Integrate with Travis CI

*Edit by @becdetat*
To run the tests:

```
npm run docker:up
npm run test:e2e-dev
```

You can run a single test file like:

```
npm run test:e2e-dev profile-wizard.test.js
```

Note that if you have the a8c VPN client installed you need to be connected to the VPN for tests to work correctly.
*end edit* 

### Changelog Note:

Dev: Implement initial E2E test suite.

### CODE REVIEW

**Comment by becdetat** [comment]
I used this to set up and run the tests:

```
npm run docker:up
npm run test:e2e-dev
```

This is awesome, I'm very excited about having e2e tests.

I had some failures when running the test - are they expected? https://gist.github.com/becdetat/9dcea7ef49ec3cd50af7696b72b2bb3a

Some appear to be timeouts but others look like it's not waiting for something to load in.

**Comment by rrennick** [comment]
> I had some failures when running the test - are they expected?

The shipping failure cascades to the checkout tests failing. I've had that happen intermittently in my environment as well. It was doing that with the core tests before I did the work on creating the component so I don't think it's related to the component. It's quite possible the Chromium for linux and Mac have subtle differences. The shipping test failure is something I'll be able to look at in the next few weeks.

File: tests/e2e/specs/front-end/front-end-cart.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createSimpleProduct } from '../../utils/components';
+import { CustomerFlow, StoreOwnerFlow } from '../../utils/flows';
+import { uiUnblocked } from '../../utils';
+
+describe( 'Cart page', () => {
```
[the remark this answers is not in the document]
Comment (rrennick) [reply tests/e2e/specs/front-end/front-end-cart.test.js:12]: > Are these tests not included anywhere else yet?

These are copied from WC core but not used anywhere else yet. I'm hoping to start on a component for these next week.

**Comment by rrennick** [comment]
> When running npm run docker:up it says 8084 is already in use. Running npm run docker:down beforehand didn't seem to help.

@joshuatf Making the port configurable is on the list to have before the component is `1.0.0` ready.

**Comment by jeffstieler** [comment]
What are the testing instructions?

I ran the same commands as @becdetat:
```sh
npm run docker:up
npm run test:e2e-dev
```

All test suites failed with:
> FAIL ../../../tests/e2e/specs/wp-admin/wp-admin-settings-tax.test.js
>   ● Test suite failed to run
> 
>     The leaking detection mechanism requires the "weak" package to be installed and work. Please install it as a dependency on your main project
> 
>       at new _default (../../jest-environment-node/build/index.js:93:13)
>       at new PuppeteerEnvironment (../../jest-environment-puppeteer/lib/PuppeteerEnvironment.js:26:1)
> 
> Test Suites: 11 failed, 11 total

**Comment by rrennick** [comment]
> All test suites failed with:

I haven't had a chance to dig into why it started requiring `weak` over the weekend. On Monday, I tried to add `weak` to the `devDependencies` and npm won't install it. Same result installing by command line.

**Comment by jeffstieler** [comment]
Ran another test here @rrennick - I think updating `@woocommerce/e2e-environment` to use `@wordpress/jest-preset-default` `^6.2.0` will fix this issue (it did when I updated manually in my local env).

**Comment by jeffstieler** [comment]
This wasn't meant to be closed by the referencing issue, whoops!

**Comment by samueljseay** [comment]
Should we close this now in favor of #5028 ?

**Comment by samueljseay** [comment]
Closing this now in favor of #5028

**Pull request closed without merging**